### PR TITLE
[Feeds] update api and store last message time

### PR DESCRIPTION
**Pull request by GuillaumeDSM**

### CODE REVIEW

File: octobot/community/feeds/abstract_feed.py [hunk 1]
```diff
@@ -41,3 +44,9 @@ async def send(self, message, channel_type, identifier, **kwargs):
 
     def can_connect(self):
         return True
+
+    def is_connected(self):
+        raise NotImplementedError("is_connected is not implemented")
+
+    def update_last_message_time(self):
```
Comment (Herklos) [inline octobot/community/feeds/abstract_feed.py:51]: 👍

**Review by Herklos — APPROVED** [review]

File: octobot/community/feeds/community_mqtt_feed.py [hunk 2]
```diff
@@ -247,9 +252,11 @@ def _on_subscribe(self, client, mid, qos, properties):
                     self.logger.error(f"Max subscription attempts reached, stopping subscription "
                                       f"to {[s.topic for s in subscriptions]}. Are you subscribing to this "
                                       f"strategy on your OctoBot account ?")
+                    self.subscribed = False
                     return
             else:
                 self._subscription_attempts = 0
+                self.subscribed = True
```
Comment (GuillaumeDSM) [inline octobot/community/feeds/community_mqtt_feed.py:259]: with this we know precisely if we are connected or not

File: octobot/community/feeds/community_mqtt_feed.py [hunk 3]
```diff
@@ -99,6 +100,9 @@ async def _stop_mqtt_client(self):
     def is_connected(self):
         return self._mqtt_client is not None and self._mqtt_client.is_connected
 
+    def is_connected_to_remote_feed(self):
```
Comment (Herklos) [inline octobot/community/feeds/community_mqtt_feed.py:103]: 👍

**Review by Herklos — APPROVED** [review]
Nice feature

**Comment by GuillaumeDSM** [comment]
requires https://github.com/Drakkar-Software/OctoBot-Trading/pull/770

File: octobot/community/authentication.py [hunk 4]
```diff
@@ -290,7 +290,28 @@ async def select_bot(self, bot_id):
         await self.on_new_bot_select()
 
     async def load_user_bots(self):
-        self.user_account.set_all_user_bots_raw_data(await self._fetch_bots())
+        self.user_account.set_all_user_bots_raw_data(
+            self._get_self_hosted_bots(
```
Comment (Herklos) [inline octobot/community/authentication.py:294]: 👍

File: octobot/community/authentication.py [hunk 5]
```diff
@@ -290,7 +290,28 @@ async def select_bot(self, bot_id):
...
+            for bot in bots
+            if self.is_self_hosted(bot)
+        ]
+
+    def is_self_hosted(self, bot):
+        deployment = bot.get(self.user_account.BOT_DEPLOYMENT)
+        if not deployment:
+            return True
+        try:
+            return deployment[self.user_account.BOT_DEPLOYMENT_TYPE] == \
```
Comment (Herklos) [inline octobot/community/authentication.py:311]: 👍

**Review by Herklos — APPROVED** [review]

**Pull request merged**